pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -906,3 +925,21 @@ func hashMapNumberOfEntries(mp *ebpf.Map, buffers *entryCountBuffers, maxRestart
 
 	return numElements
 }
+
+func isForEachElemHelperAvailable() bool {
+	return features.HaveProgramHelper(ebpf.SocketFilter, asm.FnForEachMapElem) == nil
+}
+
+func hashMapNumberOfEntriesWithHelper(mp *ebpf.Map, mapid ebpf.MapID, mphCache *mapProgHelperCache) (int64, error) {
+	prog, err := mphCache.newCachedProgramForMap(mp, mapid)
+	if err != nil {
+		return 0, err
+	}
+
+	res, _, err := prog.Test(make([]byte, 32))
```

Comment:
does this allocate each time? Could we reuse a once-allocated slice?

Reply:
I don't see it in profiles, I think it might be allocated on the stack since it's fixed size. Otherwise I could force allocate it on the stack myself to make it clearer